File: VB-Examples/BarcodeGeneration/AddLogoImageToQRCode.vb
Imports Spire.Barcode
Imports System.ComponentModel
Imports System.Text

Namespace AddLogoImageToQRCode
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim barCodeSetting As New BarcodeSettings()
			barCodeSetting.Type = BarCodeType.QRCode
			barCodeSetting.BackColor = Color.WhiteSmoke
			barCodeSetting.QRCodeDataMode = QRCodeDataMode.Byte
			barCodeSetting.QRCodeECL = QRCodeECL.M
			barCodeSetting.ShowTextOnBottom = True

			barCodeSetting.X = 3
			Dim data As String = "ABC 123456789"
			barCodeSetting.Data2D = data
			barCodeSetting.Data = data
			barCodeSetting.QRCodeLogoImage = Image.FromFile("..\..\..\..\..\..\..\Data\Logo.png")
			Dim generator As New BarCodeGenerator(barCodeSetting)
			Dim barcode As Image = generator.GenerateImage()

			Dim outputFile As String = "QRCode.png"
			barcode.Save(outputFile)
			Viewer(outputFile)
		End Sub
		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS-Examples/BarcodeGeneration/AddLogoImageToQRCode.cs
using Spire.Barcode;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace AddLogoImageToQRCode
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            BarcodeSettings barCodeSetting = new BarcodeSettings();
            barCodeSetting.Type = BarCodeType.QRCode;
            barCodeSetting.BackColor = Color.WhiteSmoke;
            barCodeSetting.QRCodeDataMode = QRCodeDataMode.Byte;
            barCodeSetting.QRCodeECL = QRCodeECL.M;
            barCodeSetting.ShowTextOnBottom = true;
           
            barCodeSetting.X = 3;
            string data = "ABC 123456789";
            barCodeSetting.Data2D = data;
            barCodeSetting.Data = data;
            barCodeSetting.QRCodeLogoImage = Image.FromFile(@"..\..\..\..\..\..\..\Data\Logo.png");
            BarCodeGenerator generator = new BarCodeGenerator(barCodeSetting);
            Image barcode = generator.GenerateImage(); ;
            string outputFile = "QRCode.png";
            barcode.Save(outputFile);
            Viewer(outputFile);
        }
        private void Viewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
